src/test/java/com/jcabi/log/VerboseProcessTest.java

```diff
@@ -310,6 +313,94 @@ public void terminatesMonitorsAndProcessIfClosedNormal() throws Exception {
     }
 
     /**
+     * VerboseProcess can run a java process which will throw a stack trace and
+     * makes sure it will group the stack trace.
+     */
+    @Test
+    public void logCompleteStackTrace() {
+        final org.apache.log4j.Logger logger =
+                org.apache.log4j.Logger.getRootLogger();
+        final TestAppender appender = new TestAppender();
+        logger.addAppender(appender);
+        final String[] commands = new String[] {
+            retrieveJavaExecLocation(), "-cp",
+            System.getProperty("java.class.path"),
+            "com.jcabi.log.VerboseProcessExample",
+        };
+        final ProcessBuilder builder = new ProcessBuilder(commands);
+        final VerboseProcess process = new VerboseProcess(
+            builder, Level.INFO, Level.SEVERE
+        );
+        IllegalArgumentException caught = new IllegalArgumentException();
+        boolean failed = false;
+        try {
+            process.stdout();
+        } catch (final IllegalArgumentException ex) {
+            caught = ex;
+            failed = true;
+        } finally {
+            logger.removeAppender(appender);
+            process.close();
+        }
+        Assert.assertTrue("Process should have failed!", failed);
+        MatcherAssert.assertThat(
```

Comment:
@dean-e-clark maybe we could perform these assertions inside `catch`? I that case both `caught` and `failed` would be unnecessary